sdk/cosmos/azure_data_cosmos/src/clients/cosmos_client.rs

```diff
@@ -77,28 +73,27 @@ impl CosmosClient {
     /// ```
     #[cfg(feature = "key_auth")]
     pub fn with_key(
-        endpoint: impl AsRef<str>,
-        key: impl Into<Secret>,
+        endpoint: &str,
+        key: Secret,
```

Comment:
This one hurt a teeny tiny bit. It means you now have to call `.into()` manually when passing a key. For example:

```rust
CosmosClient::with_key(&args.endpoint, key.into(), None)
```

I'm inclined to be fine with that, particularly since we want to discourage key auth (the API isn't even present unless you enable the `key_auth` feature in the Cosmos crate)

Reply:
Making it "harder" seems appropriate in this case.